### PR TITLE
[RLlib] Issue 9667 DDPG Torch bugs and enhancements.

### LINKED ISSUE #9667: [RLlib] Bugs in PyTorch version of DDPG
Hello,

I spotted some bugs in the implementation of DDPG in PyTorch.
* The gradient clipping is not implemented correctly, it uses the 'grad_norm_clipping' parameter instead of 'grad_clip' and the function `ray.rllib.utils.torch_ops.minimize_and_clip` which is not correct. All the others algorithms seems to use `ray.rllib.agents.a3c.a3c_torch_policy.apply_grad_clipping`. I propose to replace minimize_and_clip by the A3C function in torch_ops.

* In the PyTorch model of DDPG the parameters to bound the action space (range and minimum) are not tracked by PyTorch as they are not registered as parameters. This means that they are not converted to cuda tensors resulting in an error.

* The target model is placed on the gpu even if ray was not configure to use the gpu.

I will make a PR with everything. But I don't know if I should replace minimize_and_clip. 

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x]  I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by sven1977**
Issue 9667 describes three issues and enhancements for DDPG torch. This PR addresses all of these.

Issue #9667

Closes #9667 

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by duburcqa — APPROVED** [review]

**Comment by duburcqa** [comment]
@sven1977 Why not to fix the issue with the exploration noise I mentioned on the issue ?

File: rllib/agents/ddpg/ddpg_torch_policy.py [hunk 1]
```diff
@@ -65,9 +66,18 @@ def ddpg_actor_critic_loss(policy, model, _, train_batch):
                 mean=torch.zeros(policy_tp1.size()),
                 std=policy.config["target_noise"]), -target_noise_clip,
             target_noise_clip)
-        policy_tp1_smoothed = torch.clamp(policy_tp1 + clipped_normal_sample,
-                                          policy.action_space.low.item(0),
-                                          policy.action_space.high.item(0))
+
+        policy_tp1_smoothed = torch.min(
+            torch.max(
+                policy_tp1 + clipped_normal_sample,
```
Comment (duburcqa) [inline rllib/agents/ddpg/ddpg_torch_policy.py:72]: You must also make sure that `clipped_normal_sample` is on the same device than `policy_tp1`, since currently `clipped_normal_sample` is always on 'cpu'.

**Review by ericl — APPROVED** [review]

**Comment by sven1977** [comment]
Fixes to torch.clamp (in case action_space low/high have different values in different dimensions) in exploration/gaussian_noise and ornstein_uhlenbeck have been added. Waiting for testing.

**Comment by duburcqa** [comment]
> Fixes to torch.clamp (in case action_space low/high have different values in different dimensions) in exploration/gaussian_noise and ornstein_uhlenbeck have been added. Waiting for testing.

Nice ! Thank you

File: rllib/agents/ddpg/ddpg_torch_policy.py (hunk 1 above)
Comment (sven1977) [reply rllib/agents/ddpg/ddpg_torch_policy.py:73]: This was fixed.
